packages/gestalt/src/SideNavigation/NavigationContent.js

```diff
@@ -0,0 +1,149 @@
+import styles from '../SideNavigation.css';
+import { flattenChildrenWithKeys } from '../utils/flattenChildren';
+import { FixedZIndex } from '../zIndex';
+
+type Props = { ...SideNavigationProps };
+type Timeout = ReturnType<typeof setTimeout>;
+
+export default function NavigationContent({
+  accessibilityLabel,
+  children,
+  footer,
+  header,
+  showBorder,
+}: Props): ReactNode {
+  const navigationChildren = flattenChildrenWithKeys(children);
+
+  validateChildren({ children: navigationChildren, filterLevel: 'main' });
+
+  const {
+    collapsible,
+    collapsed: sideNavigationCollapsed,
+    onCollapse,
+    overlayPreview,
+    setOverlayPreview,
+    transitioning,
+  } = useSideNavigation();
+
+  const scrollContainer = useRef<HTMLDivElement | null>(null);
+  const [isScrolled, setIsScrolled] = useState(false);
+  const previewTimeoutRef = useRef<?Timeout>();
+
+  useEffect(() => {
+    const element = scrollContainer.current;
+    const scrollHandler = () => setIsScrolled(!!element?.scrollTop);
+
+    const mouseEnterHandler = () => {
+      if (sideNavigationCollapsed && !transitioning) {
+        clearTimeout(previewTimeoutRef.current);
+        setOverlayPreview(true);
+      }
+    };
+
+    const mouseLeaveHandler = () => {
+      if (sideNavigationCollapsed) {
+        clearTimeout(previewTimeoutRef.current);
+        previewTimeoutRef.current = setTimeout(() => setOverlayPreview(false), 1000);
+      }
+    };
+
+    element?.addEventListener('scroll', scrollHandler);
+    element?.addEventListener('mouseenter', mouseEnterHandler);
+    element?.addEventListener('mouseleave', mouseLeaveHandler);
+
+    return () => {
+      clearTimeout(previewTimeoutRef.current);
+      element?.removeEventListener('scroll', scrollHandler);
+      element?.removeEventListener('mouseenter', mouseEnterHandler);
+      element?.removeEventListener('mouseleave', mouseLeaveHandler);
+    };
+  }, [sideNavigationCollapsed, onCollapse, setOverlayPreview, transitioning]);
+
+  const isCollapsed = sideNavigationCollapsed && !overlayPreview;
+
+  const items = isCollapsed ? groupIconlessChildren(navigationChildren) : navigationChildren;
+  const iconCount = countItemsWithIcon(navigationChildren);
+
+  const shouldCollapseEmpty = iconCount === 0;
+  const shouldHideItems = sideNavigationCollapsed && shouldCollapseEmpty && !overlayPreview;
+
+  const normalWidth = 280;
+  const headerWidth = isCollapsed ? 44 : undefined;
+  const collapsedWidth = shouldCollapseEmpty ? 40 : 60;
+
+  const wrapperWidth = sideNavigationCollapsed ? collapsedWidth : normalWidth;
+  const contentWidth = isCollapsed ? collapsedWidth : normalWidth;
+
+  return (
+    <Box
+      minWidth={collapsible ? undefined : normalWidth}
+      width={collapsible ? wrapperWidth : undefined}
+      height="100%"
+      as="nav"
+      aria-label={accessibilityLabel}
+      color="default"
+      position="relative"
+      zIndex={overlayPreview ? new FixedZIndex(1) : undefined}
```

Comment:
can we foresee any   zIndex issues if it's just 1? In the examples might work but what if we have some elements below?

Reply:
this applies the whole sidenav. there is an issue of getting behind some conent which has higher z-index. But as long as the sidenav is correctly positioned, It will be displayed correctly.

I was really hesitant to use z-index here but we have no choice. We don't have a good stacking context management.